docker_explorer/de.py

```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
     """Parses the command line options."""
     self.docker_directory = os.path.abspath(options.docker_directory)
 
-  def GetContainer(self, container_id):
-    """Returns a Container object given a container_id.
+  def _GetFullContainerID(self, short_id):
+    """Searches for a container ID from its first characters.
 
     Args:
-      container_id (str): the ID of the container.
+      short_id (str): the first few characters of a container ID.
+    Returns:
+      str: the full container ID
+    Raises:
+      errors.DockerExplorerError: when we couldn't map the short version to
+        exactly one full container ID.
+    """
+    if len(short_id) == 64:
+      return short_id
+
+    containers_dir = os.path.join(self.docker_directory, 'containers')
+    possible_cids = []
+    for container_dirs in sorted(os.listdir(containers_dir)):
+      possible_cid = os.path.basename(container_dirs)
+      if possible_cid.startswith(short_id):
+        possible_cids.append(possible_cid)
+
+    possible_cids_len = len(possible_cids)
+    if possible_cids_len == 0:
+      raise errors.DockerExplorerError(
+          'Could not find any container whose ID starts with "{0}"'.format(
+              short_id))
+    elif possible_cids_len > 1:
+      raise errors.DockerExplorerError(
+          'Too many containers whose ID starts with "{0}" ({1:d})'.format(
+              short_id, possible_cids_len))
+
+    return possible_cids[0]
+
+  def GetContainer(self, container_id_part):
+    """Returns a Container object given the first characters of a container_id.
+
+    Args:
+      container_id_part (str): the first characters of a container ID.
 
     Returns:
       container.Container: the container object.
     """
+    container_id = self._GetFullContainerID(container_id_part)
```

Comment:
You don't seem to be catching any exceptions raised by `_GetFullContainerID`

Reply:
Well I do want the script to fail terribly here if we didn't find one and exactly one container ID

Reply:
I guess this comes down to choosing between catching and displaying an error to the user vs. displaying a stacktrace.

Reply:
"code is hard to write, it should be hard to use"
On a more serious note, if I catch i'll just display the error message and sys.exit(1), or return None and make a lot of checks "if container_obj" after calls to GetContainer(str).
I think having the stacktrace shown to the user is fine

Reply:
Fair enough.